### ISSUE TITLE
Rule Change: Check `a = b || a` in `logical-assignment-operators`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#17597]
### What rule do you want to change?

logical-assignment-operators

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
a = b || a
```

### What does the rule currently do for this code?

No error reported.

### What will the rule do after it's changed?

Prefer

```js
a ||= b
```

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

```js
a = b || a
```

and 

```js
a ||= b // a = a || b;
```

are not really equivalent, but most time it doesn't matter, especially when the left side has no side effect.

Maybe we can check this pattern behind an option?

**Comment by mdjermanovic** [comment]
I don't think this makes sense for the `logical-assignment-operator` rule, as this rule only suggests equivalent (or roughly equivalent) code. `a = b || a` and `a ||= b` look totally different to me. In the first case, whether or not `a` will be updated depends on the value of `b`, while in the second case it depends on the value of `a`.

```js
var a = 5, b = 6;

a = b || a;

console.log(a); // 6

```

```js
var a = 5, b = 6;

a ||= b

console.log(a); // 5

```

**Comment by fisker** [comment]
You are right, don't know what am I thinking.

It should only make sense when `a` and `b` are boolean, or only used as boolean. That will require type information, feel free to close.
